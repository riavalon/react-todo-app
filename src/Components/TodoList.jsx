import React, { Component } from 'react'
import Header from './Header'
import TodoForm from './TodoForm'
import TodoContainer from './TodoContainer'
import styles from '../Styles/TodoList.module.css'

class TodoList extends Component {
  constructor() {
    super()
    this.state = {
      items: [],
      currentItem: { text: '', key: '' },
    }
  }

  handleInput = event => {
    const itemText = event.target.value
    const currentItem = { text: itemText, key: Date.now() }
    this.setState({
      currentItem,
    })
  }

  addItem = event => {
    event.preventDefault()
    const newItem = this.state.currentItem
    if (newItem.text !== '') {
      const items = [...this.state.items, newItem]
      this.setState({
        items: items,
        currentItem: { text: '', key: '' },
      })
    }
  }
  
  deleteItem = key => {
    const filteredItems = this.state.items.filter(item => {
      return item.key !== key
    })
    this.setState({
      items: filteredItems,
    })
  }

  render() {
    return (
      <div className={styles.todoList}>
        <Header name={this.props.user} />
        <TodoForm 
          addItem={this.addItem}
          handleInput={this.handleInput}
          currentItem={this.state.currentItem}
        />
        <TodoContainer 
          todos={this.state.items} 
          onClick={this.deleteItem}
        />
      </div>
    )
  }
}
export default TodoList