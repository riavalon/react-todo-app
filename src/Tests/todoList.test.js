import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../Components/TodoList';

it('renders correctly', () => {
  const tree = renderer.create(
    <TodoList />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});