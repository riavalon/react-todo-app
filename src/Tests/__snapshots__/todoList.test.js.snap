// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="todoList"
>
  <h2
    className="heading"
  >
    's Super Cool To Do List
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Start typing here..."
      type="text"
      value=""
    />
    <button
      className="button"
      type="submit"
    >
      Add Task
    </button>
  </form>
  <ul
    className="list"
  >
    <li
      className="emptyList"
    >
      <p>
        No outstanding tasks! 
        <span
          aria-label="emojis"
          className="emojis"
          role="img"
        >
          ðŸŽ‰ðŸŽ‰
        </span>
      </p>
    </li>
  </ul>
</div>
`;
